### PR TITLE
PLATFORM-2135: Rework alert

### PR DESCRIPTION
**Pull request by ThomasPalamara**
## Proposed changes

https://lightful.atlassian.net/browse/PLATFORM-2135

All has been reviewed by design

image
image

## Types of changes

What types of changes does your code introduce to the project?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

Please use the following checklist before contributing to this repository.

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have implemented error checking and considered data not being available or loading
- [ ] I have added necessary documentation (if appropriate)
- [x] I have implemented correct type checking
- [x] I have tested my changes on a mobile layout
- [x] I have double checked with the team that I haven't introduced duplicates in functionalities (components, services, database changes, etc...)
- [x] If required, designers have signed off my changes
- [ ] I have communicated the changes to the QA team and planned testing
- [ ] I have implemented e2e tests for the new functionality

### CODE REVIEW

File: src/js/components/Alert/Alert.tsx [hunk 1]
```diff
@@ -1,26 +1,51 @@
...
+  hideIcon = false,
   ...other
 }: Props) => {
-  const classes = classNames(className, 'alert', `alert-${theme}`);
+  const classes = classNames(className, 'alert ', `alert-${theme}`);
+  const iconsByTheme = {
+    danger: 'alert',
+    success: 'approvalTick',
+    info: 'info',
+  };
```
Comment (ThomasPalamara) [inline src/js/components/Alert/Alert.tsx:32]: We can add more default icon as we go if needed

File: src/js/components/Alert/Alert.tsx [hunk 2]
```diff
@@ -1,26 +1,51 @@
...
 
   return (
     <Tag {...other} className={classes}>
-      {children}
+      {header && <h4 className="alert-heading">{header}</h4>}
+      <div className="d-flex align-items-start">
+        {!hideIcon && (iconsByTheme[theme] || icon) && (
```
Comment (ThomasPalamara) [inline src/js/components/Alert/Alert.tsx:38]: If no props icon and no default icon is set for the theme we just don't display any icon

File: src/js/components/Alert/Alert.tsx [hunk 3]
```diff
@@ -1,26 +1,51 @@
...
 };
 
 const Alert: React.FC<Props> = ({
   children,
   className,
+  header,
+  icon,
   tag: Tag = 'div',
   theme = 'primary',
+  hideIcon = false,
```
Comment (ThomasPalamara) [inline src/js/components/Alert/Alert.tsx:24]: By default, we show the icon as requested by design but if no icon is set we won't have any empty element as explained earlier

File: src/js/components/PostCard/PostCard.tsx [hunk 4]
```diff
@@ -129,17 +130,13 @@ class PostCard extends React.Component<Props, State> {
...
-                  { 'mb-1': i < errors.length - 1 }
-                )}
-                style={{ lineHeight: 1.4 }}
+                className={classNames('mx-2', { 'mb-1': i < errors.length - 1 })}
+                theme="danger"
               >
-                <Icon className="mr-half flex-shrink-0" name="alert" theme="danger" size={18} />
                 {error}
-              </div>
+              </Alert>
```
Comment (ThomasPalamara) [inline src/js/components/PostCard/PostCard.tsx:139]: Updated PostCard to use the Alert component

**Review by josephrace — APPROVED** [review]
Looks good 👍

**Pull request merged**